src/custom-element.js

```diff
@@ -328,8 +333,19 @@ function createBaseCustomElementClass(win) {
      * @package
      */
     getAmpDoc() {
-      devAssert(this.ampdoc_, 'no ampdoc yet, since element is not attached');
-      return /** @typedef {!./service/ampdoc-impl.AmpDoc} */ this.ampdoc_;
+      if (this.ampdoc_) {
+        return this.ampdoc_;
+      }
+      if (!this.isConnected_) {
+        throw dev().createError(
+          this.tagName.toLowerCase(),
+          'attempted to get ampdoc before being connected'
+        );
+      }
+      const win = toWin(this.ownerDocument.defaultView);
+      const ampdocService = Services.ampdocServiceFor(win);
+      this.ampdoc_ = ampdocService.getAmpDoc(this);
```

Comment:
I think we just need `Services.ampdoc(this)` now. Just an API refresh, so we can simplify this a bit.

Reply:
Done.